### ISSUE TITLE
Optimization for ArrayBuffer / DataView? Websockets.

### ISSUE THREAD

**Issue by vans163** [pierrec/node-lz4#46]
On the browser side when you binary data off websockets you get it as an ArrayBuffer or Blob.

Right now you cannot pass that to new Buffer() constructor.

Instead you need to convert the buffer to a Uint8Array, this conversion is very expensive for some reason.

Then once you get your output, you need to convert it again to a Uint8Array if you want to use it for something like canvas element; ctx.putImageData(imageData, 0, 0) for example.

**Comment by pierrec** [comment]
Yes support for more (better) input types is required. Unfortunately, time is scarce :(.
Pull requests anyone?
